test/realm_test.dart

```diff
@@ -1567,6 +1566,75 @@ Future<void> main([List<String>? args]) async {
     //test the realm can be opened.
     final realm = getRealm(Configuration.disconnectedSync([Product.schema], path: path, encryptionKey: key));
   });
+
+  test('Realm.refresh', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    final result = realm.refresh();
+    expect(result, true);
+  });
+
+  test('Realm.onRefresh sync transaction', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    var called = false;
+    final transaction = realm.beginWrite();
+    realm.refreshAsync().then((_) => called = true);
+    realm.add(Person("name"));
+    transaction.commit();
+
+    await Future<void>.delayed(Duration(milliseconds: 1));
```

Comment:
Is it correct to relly on the delay for 1 milisecond to ensure that the `refreshAsync` is completed?

Reply:
yes, this makes sure to release the event loop.